internal/backends/winit/renderer/femtovg/glcontext.rs

```diff
@@ -49,144 +51,108 @@ impl OpenGLContext {
-                use cocoa::appkit::NSView;
-                let view_id: cocoa::base::id = ns_view as *const _ as *mut _;
-                unsafe {
-                    NSView::setLayerContentsPlacement(view_id, cocoa::appkit::NSViewLayerContentsPlacement::NSViewLayerContentsPlacementTopLeft)
-                }
-            }
+                .expect("internal error: Could not find any matching GL configuration")
+            })?;
 
-            Self { context: not_current_gl_context.make_current(&surface).unwrap(), surface }
-        }
+        let gl_display = gl_config.display();
+
+        let raw_window_handle = window.as_ref().map(|w| w.raw_window_handle());
+
+        let gles_context_attributes = ContextAttributesBuilder::new()
+            .with_context_api(ContextApi::Gles(Some(glutin::context::Version {
+                major: 2,
+                minor: 0,
+            })))
+            .build(raw_window_handle);
+
+        let fallback_context_attributes = ContextAttributesBuilder::new().build(raw_window_handle);
 
-        #[cfg(target_arch = "wasm32")]
+        let not_current_gl_context = unsafe {
+            gl_display
+                .create_context(&gl_config, &gles_context_attributes)
+                .or_else(|_| gl_display.create_context(&gl_config, &fallback_context_attributes))?
+        };
+
+        let window = match window {
+            Some(window) => window,
+            None => glutin_winit::finalize_window(window_target, window_builder, &gl_config)?,
+        };
+
+        let size: winit::dpi::PhysicalSize<u32> = window.inner_size();
+
+        let width: std::num::NonZeroU32 =
+            size.width.try_into().expect("new context called with zero width window");
+        let height: std::num::NonZeroU32 =
+            size.height.try_into().expect("new context called with zero height window");
+
+        let attrs = SurfaceAttributesBuilder::<WindowSurface>::new().build(
+            window.raw_window_handle(),
+            width,
+            height,
+        );
+
+        let surface = unsafe { gl_display.create_window_surface(&gl_config, &attrs)? };
+
+        #[cfg(target_os = "macos")]
+        if let raw_window_handle::RawWindowHandle::AppKit(raw_window_handle::AppKitWindowHandle {
+            ns_view,
+            ..
+        }) = window.raw_window_handle()
         {
-            use wasm_bindgen::JsCast;
-
-            let canvas = web_sys::window()
-                .unwrap()
-                .document()
-                .unwrap()
-                .get_element_by_id(canvas_id)
-                .unwrap()
-                .dyn_into::<web_sys::HtmlCanvasElement>()
-                .unwrap();
-
-            Self { canvas }
+            use cocoa::appkit::NSView;
+            let view_id: cocoa::base::id = ns_view as *const _ as *mut _;
+            unsafe {
+                NSView::setLayerContentsPlacement(view_id, cocoa::appkit::NSViewLayerContentsPlacement::NSViewLayerContentsPlacementTopLeft)
```

Comment:
Why do we do that? shouldn't glutin or winit or glutin-winit  take care of that? This probably deserve a comment

Reply:
This is to reduce "jitter" when resizing on macOS. This is somewhat of a preference. But good point about the comment, I'll resurrect that from the original commit long time ago :)